designcompose/src/main/java/com/android/designcompose/DesignView.kt

```diff
@@ -817,6 +817,11 @@ enum class LiveUpdateMode {
     OFFLINE // Live updates off (load from serialized file)
 }
 
+class DesignComposeCallbacks(
+    val docReadyCallback: ((String) -> Unit)? = null,
+    val docUpdateCallback: ((ByteArray?) -> Unit)? = null,
```

Comment:
Maybe via DesignSettings.

Reply:
I mean, yea this is a bit clunky, I was trying to throw something together quickly for Umesh, but also, if you don't want to get serialized file updates every time it updates, then just don't register for the callback. If we want a callback to just inform you that there has been an update, we can add that functionality or rename this callback.

Reply:
I still feel that serializing the file shouldn't happen automatically as part of the library calling the callback. It should be the code that registers the callback that makes requests for further work to be done by the library.
And yeah, I think it'd be better to handle registering of listeners via DesignSettings, and that's where we could have the function to request the serialized file.

Reply:
Rob, I am fine with this change since the data is not fetched if the callback is null.  What you have currently is an alternate form for registering callbacks - like with add/remove way via DesignSettings.

Reply:
I did explore using DesignSettings when I started this and it's not that simple. The reason I did it this way is that the generated code knows the ID of the Figma document you care about, and that's needed so that your callback is called on the correct document getting updated.

I can also call this function something like `newDocDataCallback`

Reply:
Alright, overruled. I still want to have it added to the documentation though.

Reply:
Let me think about this a bit over the weekend, and I'll also add some documentation when I update this.